### DISCUSSION TITLE
Can I use webdriverio for e2e testing with single-spa ?

### DISCUSSION THREAD

**Discussion by anhhai680** [webdriverio/webdriverio#7374] [Q&A]
I'm developing a `single-spa` application and I'm trying to use webdriverio for e2e testing with `single-spa` application but unfortunately, it didn't work well. It couldn't found html elements of my application after it mounted application to root-config application.

Does anyone have more experience about `single-spa` and e2e testing on this framework ? Please help me and give me any solution to resolve this issue. Thank you in advance to everyone!

[https://single-spa.js.org/docs/examples](https://single-spa.js.org/docs/examples)

**Comment by christian-bromann** [comment]
Hey @anhhai680 👋 

> It couldn't found html elements of my application after it mounted application to root-config application.

Can you explain a bit further, maybe with a reproducible example? WebdriverIO usually tries multiple times until it finds the element even though it appears a bit later after page load.
